### ISSUE TITLE
Disabling autoinstall globally

### ISSUE THREAD

**Issue by mischnic** [parcel-bundler/parcel#2081]
# 🙋 feature request

Add the possibility to set `--no-autoinstall` globally using a env variable (e.g. `PARCEL_NO_AUTOINSTALL`).

## 💁 Possible Solution

Make it so. I can submit a PR if you would accept this feature.

## 🔦 Context

The autoinstall feature might be great for beginners, but today, it worked against me two times in a row:
- I switched over from React to Preact. First, I uninstalled react and installed the preact package to then start parcel and see where I need to change the imports. But instead, parcel reinstalled react.
- I imported "style.css" instead of "./style.css", and parcel installed the "style.css" package.
I don't want to add the `--no-autoinstall` flag to every project of mine,

**Comment by cacheflow** [comment]
PR for this if anyone has a chance: https://github.com/parcel-bundler/parcel/pull/2130

**Comment by chocolateboy** [comment]
My 2c: installing anything without the user's express knowledge and consent is a huge security risk which will end up compromising someone's system and bringing this project into disrepute.

While this workaround is well-meaning, and arguably better than the status quo, it's still a sticking plaster, which legitimizes the current (mis)behavior rather than fixing the underlying problem: *it should be opt in rather than opt out*.

**Comment by DeMoorJasper** [comment]
@chocolateboy i agree about installing code dependencies, but not about the parsers and such parcel internally uses to process assets as it removes some of the complexity without the need of including them all as parcel dependencies

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by mischnic** [comment]
Possible example for the security risk: With `import React from 'reac';`, parcel installs that dependency (and runs it's postinstall scripts). Without autoinstall, I would have realised my typo while typing `yarn add reac`.

**Comment by chocolateboy** [comment]
Related: https://github.com/zkat/npx/issues/198
